test/distributed/org/apache/cassandra/distributed/test/SSTableIdGenerationTest.java

```diff
@@ -0,0 +1,229 @@
+                                                .stream()
+                                                .map(sstr -> sstr.descriptor)
+                                                .collect(Collectors.toSet());
+                assertThat(descs.stream().filter(desc -> desc.generation instanceof SequenceBasedSSTableId).count()).isEqualTo(expectedSeqGenIds);
+                assertThat(descs.stream().filter(desc -> desc.generation instanceof UUIDBasedSSTableId).count()).isEqualTo(expectedUUIDGenIds);
+            });
+        }
+        catch (Throwable t)
+        {
+            throw new AssertionError(t);
+        }
+    }
+
+    @Test
+    public void testRestartWithUUIDEnabled() throws IOException
+    {
+        try (Cluster cluster = init(Cluster.build(1).withConfig(config -> config.set(ENABLE_UUID_FIELD_NAME, false)).start()))
+        {
+            cluster.disableAutoCompaction(KEYSPACE);
+            cluster.schemaChange("CREATE TABLE " + KEYSPACE + ".tbl (pk int, v int, primary key (pk))");
+            createSSTable(cluster, "tbl");
+            createSSTable(cluster, "tbl");
+            assertSSTablesCount(cluster, "tbl", 2, 0);
+            verfiySSTableActivity(cluster, true);
+
+            waitOn(cluster.get(1).shutdown());
+            cluster.get(1).config().set(ENABLE_UUID_FIELD_NAME, true);
+            cluster.get(1).startup();
+
+            createSSTable(cluster, "tbl");
+            createSSTable(cluster, "tbl");
+
+            assertSSTablesCount(cluster, "tbl", 2, 3);
+
+            assertThat(cluster.get(1).executeInternalWithResult(String.format("SELECT * FROM %s.%s", KEYSPACE, "tbl")).toObjectArrays()).hasSize(4);
+        }
+    }
+
+    @Test
+    public void testRestartWithUUIDDisabled() throws IOException
+    {
+        try (Cluster cluster = init(Cluster.build(1).withConfig(config -> config.set(ENABLE_UUID_FIELD_NAME, true)).start()))
+        {
+            cluster.disableAutoCompaction(KEYSPACE);
+            cluster.schemaChange("CREATE TABLE " + KEYSPACE + ".tbl (pk int, v int, primary key (pk))");
+            createSSTable(cluster, "tbl");
+            createSSTable(cluster, "tbl");
+            assertSSTablesCount(cluster, "tbl", 0, 2);
+            verfiySSTableActivity(cluster, false);
+
+            waitOn(cluster.get(1).shutdown(true));
+            cluster.get(1).config().set(ENABLE_UUID_FIELD_NAME, false);
+
+            Assertions.assertThatExceptionOfType(RuntimeException.class)
+                      .isThrownBy(() -> cluster.get(1).startup())
+                      .withCauseInstanceOf(SystemExitException.class);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testCompactionStrategiesWithMixedSSTables() throws Exception
+    {
+        testCompactionStrategiesWithMixedSSTables(SizeTieredCompactionStrategy.class, DateTieredCompactionStrategy.class, TimeWindowCompactionStrategy.class, LeveledCompactionStrategy.class);
```

Comment:
Nit: can we break this long line?
```suggestion
        testCompactionStrategiesWithMixedSSTables(SizeTieredCompactionStrategy.class,
                                                  DateTieredCompactionStrategy.class,
                                                  TimeWindowCompactionStrategy.class,
                                                  LeveledCompactionStrategy.class);
```